mlflow/keras.py

```diff
@@ -759,26 +762,24 @@ def _run_and_log_function(self, original, args, kwargs, unlogged_params, callbac
 
         return history
 
-    def fit(self, *args, **kwargs):
-        original = gorilla.get_original_attribute(keras.Model, "fit")
+    def fit(original, self, *args, **kwargs):
```

Comment:
Consider having this sort of logic extracted out to the autolog wrapper:

```
        if not mlflow.active_run():
            try_mlflow_log(mlflow.start_run)
            auto_end_run = True
        else:
            auto_end_run = False
```

Reply:
Done!